dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1773,4 +1826,668 @@ public void getDirectEpersonGroups() throws Exception {
+        assertTrue(StringUtils.equalsIgnoreCase(oldPassword.getHashString(),newPasswordHash.getHashString()));
+        assertFalse(registrationDataService.findByEmail(context, ePerson.getEmail()) == null);
+        assertTrue(StringUtils.equals(registrationDataService.findByEmail(context, ePerson.getEmail()).getToken(),
+                                      tokenForEPerson));
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.deleteByToken(context, tokenForEPerson);
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void patchReplacePasswordWithOtherUserTokenFail() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("Johndoe@fake-email.com")
+                                        .withPassword(password)
+                                        .build();
+
+
+        EPerson ePersonTwo = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("Smith", "Donald")
+                                        .withEmail("donaldSmith@fake-email.com")
+                                        .withPassword(password)
+                                        .build();
+
+        String newPassword = "newpassword";
+
+        context.restoreAuthSystemState();
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/password", newPassword);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+        accountService.sendRegistrationInfo(context, ePerson.getEmail());
+        accountService.sendRegistrationInfo(context, ePersonTwo.getEmail());
+        String tokenForEPerson = registrationDataService.findByEmail(context, ePerson.getEmail()).getToken();
+        String tokenForEPersonTwo = registrationDataService.findByEmail(context, ePersonTwo.getEmail()).getToken();
+
+        String token = getAuthToken(admin.getEmail(), password);
+        PasswordHash oldPassword = ePersonService.getPasswordHash(ePerson);
+        // updates password
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON)
+                                     .param("token", tokenForEPersonTwo))
+                        .andExpect(status().isForbidden());
+
+        PasswordHash newPasswordHash = ePersonService.getPasswordHash(ePerson);
+        assertTrue(StringUtils.equalsIgnoreCase(oldPassword.getHashString(),newPasswordHash.getHashString()));
+        assertFalse(registrationDataService.findByEmail(context, ePerson.getEmail()) == null);
```

Comment:
Again, can be `assertEquals` and `assertNotNull`

Reply:
This has been adjusted